nil/client/direct_client.go

```diff
@@ -127,7 +149,29 @@ func (c *DirectClient) GetBlocksRange(
 	fullTx bool,
 	batchSize int,
 ) ([]*jsonrpc.RPCBlock, error) {
-	panic("Not supported")
+	if from >= to {
+		return nil, nil
+	}
+
+	result := make([]*jsonrpc.RPCBlock, 0)
+	for curBlockId := from; curBlockId < to; curBlockId += types.BlockNumber(batchSize) {
```

Comment:
```suggestion
	for curBlockId := from; curBlockId < to; curBlockId ++ {
```
You can remove nested loop.

This part is relevant for http queries but here you don't send anything over network.

Reply:
fixed